src/llama.cpp

```diff
@@ -3258,6 +3258,11 @@ static void llama_kv_cache_seq_add(
 
     if (p0 < 0) p0 = 0;
     if (p1 < 0) p1 = std::numeric_limits<llama_pos>::max();
+    // If there is no range then return early to avoid looping over the cache.
+    if (p0 == p1) {
+        cache.head = 0;
```

Comment:
Is `cache.head = 0` necessary here?

Reply:
I'm not sure, but since it is updated before this function returns I thought I would be better to make sure that it is updated. But I'll take a closer look to verify if this is required. Thanks

Reply:
I checked and the setting of `cache.head` is unnecessary. I've update now :+1: